src/transforms/repair.jl

```diff
@@ -171,3 +173,60 @@ function _stretch10(g::Geometry)
   T = numtype(lentype(g))
   Stretch(ntuple(i -> one(T) + 10atol(T), embeddim(g)))
 end
+
+# ---------------
+# OPERATION (11)
+# ---------------
+
+function apply(::Repair{11}, poly::PolyArea)
+  r = rings(poly)
+  outer = r[begin]
+  inners = hasholes(poly) ? r[(begin + 1):end] : eltype(r)[]
+
+  # original orientations
+  oouter = orientation(outer)
+  oinners = orientation.(inners)
+  
+  # adjust orientation
+  oadjust(r, oold, onew) = oold == onew ? r : reverse(r)
+  outer = oadjust(outer, oouter, CCW)
+  inners = oadjust.(inners, oinners, CW)
+
+  PolyArea([outer; inners]), (oouter, oinners)
+end
+
+function revert(::Repair{11}, poly::PolyArea, cache)
+  r = rings(poly)
+  outer = r[begin]
+  inners = hasholes(poly) ? r[(begin + 1):end] : eltype(r)[]
+
+  # revert to original orientation
+  (oouter, oinners) = cache
+  oadjust(r, oold, onew) = oold == onew ? r : reverse(r)
+  outer = oadjust(outer, CCW, oouter)
+  inners = oadjust.(inners, CW, oinners)
+
+  PolyArea([outer; inners])
+end
+
+# ---------------
+# OPERATION (12)
+# ---------------
+
+function apply(::Repair{12}, poly::PolyArea)
+  r = rings(poly)
+  outer = r[begin]
+  inners = hasholes(poly) ? r[(begin + 1):end] : eltype(r)[]
+
+  # fix degeneracy
+  if nvertices(outer) == 2
+    A, B = vertices(outer)
+    P = center(Segment(A, B))
+    outer = Ring(A, P, B)
+  end
+
+  # remove degenerated rings
+  inners = filter(r -> nvertices(r) > 2, inners)
+
+  PolyArea([outer; inners]), nothing
```

Comment:
try to organize this in two sections, one that fixes the outer ring, and another that filters the inner rings. the variables can be defined in each section instead of in the beginning of the function